api/types/kustomization.go

```diff
@@ -332,3 +359,101 @@ func (k *Kustomization) Unmarshal(y []byte) error {
 	*k = nk
 	return nil
 }
+
+// splitPatchStrategicMerge splits a single PatchStrategicMerge file into multiple PatchStrategicMerge files, if the
+// file contains multiple documents separated by the yaml separator.
+func splitPatchStrategicMerge(p PatchStrategicMerge, fSys filesys.FileSystem) ([]PatchStrategicMerge, error) {
+	patchContentBytes, err := fSys.ReadFile(string(p))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %v: %w", p, err)
+	}
+
+	splitPatchContent, err := kio.SplitDocuments(string(patchContentBytes))
+	if err != nil {
+		return nil, fmt.Errorf("failed to split patch file %v: %w", p, err)
+	}
+
+	// If the split resulted in only one document, there is nothing to do, so just return the original patch without any changes.
+	if len(splitPatchContent) == 1 {
+		return []PatchStrategicMerge{p}, nil
+	}
+
+	// Find the new patches, removing any empty ones.
+	var newPatches []string
+	for _, pc := range splitPatchContent {
+		trimmedPatchContent := strings.TrimSpace(pc)
+		if len(trimmedPatchContent) > 0 {
+			newPatches = append(newPatches, trimmedPatchContent+"\n")
+		}
+	}
+
+	// If there is only one new patch, that means there was one or more empty ones that were discarded. In this case,
+	// overwrite the original patch file with the new patch content and return it.
+	if len(newPatches) == 1 {
+		err := fSys.WriteFile(string(p), []byte(newPatches[0]))
+		if err != nil {
+			return nil, fmt.Errorf("failed to write file %v: %w", p, err)
+		}
+		return []PatchStrategicMerge{p}, nil
+	}
+
+	// If there are multiple new patches, create new patch files for each one, remove the original patch file, and
+	// return the list of new patch files.
+	result := make([]PatchStrategicMerge, len(newPatches))
+	for i, newPatchContent := range newPatches {
+		newPatchPath, err := availableFilename(string(p), i+1, fSys)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find available filename for %v: %w", p, err)
+		}
+		err = fSys.WriteFile(newPatchPath, []byte(newPatchContent))
+		if err != nil {
+			return nil, fmt.Errorf("failed to write file %v: %w", newPatchPath, err)
+		}
+		result[i] = PatchStrategicMerge(newPatchPath)
+	}
+
+	err = fSys.RemoveAll(string(p))
+	if err != nil {
+		return nil, fmt.Errorf("failed to remove file %v: %w", p, err)
+	}
+
+	return result, nil
+}
+
+// splitInlinePatchStrategicMerge splits a single inline PatchStrategicMerge into multiple inline PatchStrategicMerges,
+// if it contains multiple documents separated by the yaml separator.
+func splitInlinePatchStrategicMerge(p PatchStrategicMerge) ([]PatchStrategicMerge, error) {
```

Comment:
nit: this method above starts with near-identical content. Let's name this something more generic like `splitPatchDocuments` and reuse it

Reply:
I renamed both of the functions to have more appropriate names, I think.  Let me know what you think:
* `splitPatchFile`
* `splitInlinePatch`